Design new .NET Core Setup/Install area; Rewrite prereqs articles; Add package managers (#15569)
dotnet/docs: docs/core/setup/install-sdk.md

---
title: Install .NET Core SDK on Windows, Linux, and macOS - .NET Core
description: Learn how to install .NET Core on Windows, Linux, and macOS. Discover the dependencies required to develop .NET Core apps.
author: thraka
ms.author: adegeo
ms.date: 11/06/2019
ms.custom: "updateeachrelease"
zone_pivot_groups: operating-systems-set-one
---

# Install the .NET Core SDK

> In this article, you'll learn how to download and install the .NET Core SDK. The .NET Core SDK is used to create .NET Core apps and libraries. The .NET Core runtime is always installed with the SDK.

Reviewer: We don't really teach people how to download 

```suggestion
In this article, you'll learn how to install the .NET Core SDK. The .NET Core SDK is used to create .NET Core apps and libraries. The .NET Core runtime is always installed with the SDK.
```

You can download and install .NET Core directly with one of the following links:

> - [.NET Core 3.1 Preview 3 downloads](https://dotnet.microsoft.com/download/dotnet-core/3.1)

Reviewer: wouldn't this be install with an installer then?

Should you move there?

Author: I honestly just added it here for a shortcut of the "latest" preview. These download pages though go to the .NET website pages, which list all ways (including package managers) in a nice table matrix format. People get annoyed when they have to go down sections to find the right download link, I prefer to have a simple list at the start of the article to (when they're impatient) get them right to a download page.

- [.NET Core 3.0 downloads](https://dotnet.microsoft.com/download/dotnet-core/3.0)
- [.NET Core 2.2 downloads](https://dotnet.microsoft.com/download/dotnet-core/2.2)
- [.NET Core 2.1 downloads](https://dotnet.microsoft.com/download/dotnet-core/2.1)

You can also install .NET Core as part of an integrated development environment (IDE), detailed in the sections below.

## Install with an installer

Both Windows and macOS have standalone installers that can be used to install the .NET Core SDK.

- Windows [x64 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-windows-x64-installer) | [x32 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-windows-x86-installer)
- macOS [x64 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-macos-x64-installer)

::: zone pivot="os-linux"

## Install with a package manager

> You can install .NET Core with many of the common linux package managers. For more information, see [Linux Package Manager - Install .NET Core](linux-package-manager-rhel7.md).

Reviewer: Make sure there are no lowercase Linux throughout 

```suggestion
You can install the .NET Core SDK with many of the common Linux package managers. For more information, see [Linux Package Manager - Install .NET Core](linux-package-manager-rhel7.md).
```

::: zone-end

::: zone pivot="os-windows"

> ## Visual Studio

Reviewer: This one is the only one without a verb

```suggestion
## Install with Visual Studio
```

> If you're using Visual Studio to develop .NET Core apps, the following table describes the minimum required version of Visual Studio based on the target .NET Core runtime.

Reviewer 2: ```suggestion
If you're using Visual Studio to develop .NET Core apps, the following table describes the minimum required version of Visual Studio based on the target .NET Core SDK version.
```

| .NET Core SDK version | Visual Studio version                      |
| --------------------- | ------------------------------------------ |
| 3.0                   | Visual Studio 2019 version 16.3 or higher. |
| 2.2                   | Visual Studio 2017 version 15.9 or higher. |
| 2.1                   | Visual Studio 2017 version 15.7 or higher. |

If you already have Visual Studio installed, you can check your version with the following steps.

01. Open Visual Studio.

> 01. Select the **Help** menu > **About Microsoft Visual Studio**.

Reviewer 2: ```suggestion
01. Select **Help** > **About Microsoft Visual Studio**.
```

01. Read the version number from the **About** dialog.

### Download

Visual Studio installs the latest .NET Core SDK and runtime, so you can target your app to run on the .NET Core runtime version of your choice.

- [Download Visual Studio](https://www.visualstudio.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=button+cta&utm_content=download+vs2019).

> Selecting any of the managed workloads for **Web**, **Desktop**, or **Azure development**, will install the .NET Core SDK development tools and runtime. Specifically choosing the **.NET Core** workload also installs .NET Core.

Reviewer 2: I think it's best to explicitly state the workloads here, e.g. **.NET Core cross-platform development**.

Reviewer: I agree. We had that earlier with the specific workload names and sections.

https://docs.microsoft.com/en-us/dotnet/core/windows-prerequisites?tabs=netcore30#prerequisites-to-develop-net-core-apps-with-visual-studio

[![Windows Visual Studio 2019 with .NET Core workload](media/install-sdk/windows-install-visual-studio-2019.png)](media/install-sdk/windows-install-visual-studio-2019.png#lightbox)

::: zone-end

::: zone pivot="os-macos"

## Install with Visual Studio for Mac

> Visual Studio for Mac installs the .NET Core SDK when the **.NET Core workload** is selected. To get started with .NET Core development on macOS, see [Install Visual Studio 2019 for Mac](/visualstudio/mac/installation).

Reviewer 2: ```suggestion
Visual Studio for Mac installs the .NET Core SDK when the **.NET Core** workload is selected. To get started with .NET Core development on macOS, see [Install Visual Studio 2019 for Mac](/visualstudio/mac/installation).
```

[![macOS Visual Studio 2019 for Mac with .NET Core workload feature](media/install-sdk/mac-install-selection.png)](media/install-sdk/mac-install-selection.png#lightbox)

::: zone-end

> ## Install with Visual Studio Code

Reviewer: I still feel this is a bit misleading. You're not installing with VS Code, you're using an installer. But I understand the intent behind this.

> Visual Studio Code is a powerful and lightweight source code editor that runs on your desktop. VS Code is available for Windows, macOS, and Linux.

Reviewer 2: ```suggestion
Visual Studio Code is a powerful and lightweight source code editor that runs on your desktop. Visual Studio Code is available for Windows, macOS, and Linux.
```

I don't think it's every supposed to be abbreviated in docs.

Reviewer: No, but I think Greg said he got agreement to shorten after first usage. 
I'd just avoid trouble with branding and use the full name.

> While VS Code doesn't come with .NET Core support, adding .NET Core support is simple.

Reviewer 2: ```suggestion
While Visual Studio Code doesn't come with .NET Core support, adding .NET Core support is simple.
```

01. [Download and install Visual Studio Code](https://code.visualstudio.com/Download).
01. [Download and install the .NET Core SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0).

> 01. [Install the C# extension from the VS Code marketplace](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp).

Reviewer 2: Expand VS Code.

::: zone pivot="os-windows"

## Install with PowerShell automation

The [dotnet-install scripts](../tools/dotnet-install-script.md) are used for automation and non-admin installs of the SDK. You can download the script from the [dotnet-install script reference page](../tools/dotnet-install-script.md).

> The script defaults to installing the latest [long term support (LTS)](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) version, which is currently .NET Core 2.1. To install the current release of .NET Core (3.0), run the script with the following switch:

Reviewer 2: ```suggestion
The script defaults to installing the latest [long term support (LTS)](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) version, which is currently .NET Core 2.1. To install the current release of .NET Core, run the following script and substitute the version as necessary:
```

```powershell
dotnet-install.ps1 -Channel 3.0
```

::: zone-end

::: zone pivot="os-linux,os-macos"

## Install with bash automation

The [dotnet-install scripts](../tools/dotnet-install-script.md) are used for automation and non-admin installs of the SDK. You can download the script from the [dotnet-install script reference page](../tools/dotnet-install-script.md).

> The script defaults to installing the latest [long term support (LTS)](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) version, which is currently .NET Core 2.1. To install the current release of .NET Core (3.0), run the script with the following switch:

Reviewer 2: ```suggestion
The script defaults to installing the latest [long term support (LTS)](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) version, which is currently .NET Core 2.1. To install the current release of .NET Core, run the script with the following switch:
```

```bash
./dotnet-install.sh -c Current
```

::: zone-end

## Docker

> Containers provide a lightweight way to isolate your application from the rest of the host system, sharing just the kernel, and using resources given to your application.

Reviewer 2: Same suggestion as before for rewording this a bit.

.NET Core can run in a Docker container. Official .NET Core Docker images are published to the Microsoft Container Registry (MCR) and are discoverable at the [Microsoft .NET Core Docker Hub repository](https://hub.docker.com/_/microsoft-dotnet-core/). Each repository contains images for different combinations of the .NET (SDK or Runtime) and OS that you can use.

Microsoft provides images that are tailored for specific scenarios. For example, the [ASP.NET Core repository](https://hub.docker.com/_/microsoft-dotnet-core-aspnet/) provides images that are built for running ASP.NET Core apps in production.

For more information about using .NET Core in a Docker container, see [Introduction to .NET and Docker](../docker/introduction.md) and [Samples](https://github.com/dotnet/dotnet-docker/blob/master/samples/README.md).

> ## Next steps

Reviewer: On MVC land (https://review.docs.microsoft.com/en-us/help/contribute/contribute-how-to-mvc-tutorial?branch=master), next steps is a single link. Should we use See also here?

```suggestion
## See also
```

Author: I do prefer **See also**, but I dont know if we're supposed to use it? All the guidance I've seen has deemed **Next steps** as the header title. That MVC suggestion you pointed to though is *only* for the tutorial type. Other pages can have more. For example the overview type says 1-3 next steps.

---
title: Install .NET Core SDK on Windows, Linux, and macOS
description: Learn how to install .NET Core on Windows, Linux, and macOS. Discover the dependencies required to develop .NET Core apps.
author: thraka
ms.author: adegeo
ms.date: 11/06/2019
zone_pivot_groups: operating-systems-set-one

> ---

Reviewer: make sure you have the ms.custom metadata for us to review this article for each vs release like the windows article had

# Install the .NET Core SDK

In this article you'll learn how to download and install the .NET Core SDK. The .NET Core SDK is used to create create .NET Core apps and libraries. The .NET Core runtime is always installed with the SDK.

> Here are links to the .NET Core downloads page by version:

Author: We're still figuring out how we want to present the download links and what from the original site will be deleted.

- [.NET Core 3.0 downloads](https://dotnet.microsoft.com/download/dotnet-core/3.0).
- [.NET Core 2.2 downloads](https://dotnet.microsoft.com/download/dotnet-core/2.2).
- [.NET Core 2.1 downloads](https://dotnet.microsoft.com/download/dotnet-core/2.1).

## Install with an IDE

When you install a supported IDE, you can also install .NET Core. By keeping the IDE up-to-date, you'll receive the latest supported .NET Core SDK.

::: zone pivot="os-windows"

> - Download [Visual Studio](https://www.visualstudio.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=button+cta&utm_content=download+vs2019)\

Reviewer: you need to do more than just install VS to get the SDK installed. This is only explained in the other section. Why this is not combined?

For more information, see [Visual Studio](#visual-studio).

- Download [Visual Studio Code](https://code.visualstudio.com/)\
For more information, see [Install with Visual Studio Code](#install-with-visual-studio-code).

::: zone-end

::: zone pivot="os-linux"

- Download [Visual Studio Code](https://code.visualstudio.com/)\
For more information, see [Install with Visual Studio Code](#install-with-visual-studio-code).

::: zone-end

::: zone pivot="os-macos"

- Download [Visual Studio for Mac](https://visualstudio.microsoft.com/vs/mac/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)\
For more information, see [Install with Visual Studio for Mac](#install-with-visual-studio-for-mac).

- Download [Visual Studio Code](https://code.visualstudio.com/)\
For more information, see [Install with Visual Studio Code](#install-with-visual-studio-code).

::: zone-end

## Install with an installer

Both Windows and macOS have standalone installers which can be used to install the .NET Core SDK.

> - Windows [x64 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-windows-x64-installer) | [x32 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-windows-x86-installer)

Reviewer: these links get outdated fast. You already talked in the introduction where to find the downloads. One of them should go.

- macOS [x64 CPUs](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-macos-x64-installer)

::: zone pivot="os-linux"

## Install with a package manager

> If you want to install .NET Core, either SDK or runtime, use the [Linux Package Manager](linux-package-manager-rhel7.md) page.

Reviewer: This page is about the SDK right? So why mention the runtime?

::: zone-end

::: zone pivot="os-windows"

## Visual Studio

If you're using Visual Studio to develop .NET Core apps, the following table describes the minimum required version of Visual Studio based on the target .NET Core runtime.

| .NET Core SDK version | Visual Studio version                      |
| --------------------- | ------------------------------------------ |
| 3.0                   | Visual Studio 2019 version 16.3 or higher. |
| 2.2                   | Visual Studio 2017 version 15.9 or higher. |
| 2.1                   | Visual Studio 2017 version 15.7 or higher. |

If you already have Visual Studio installed, you can check your version with the following steps.

> 01. Open Visual Studio.

Reviewer: I personally dislike double digits when you have less than 10 items

```suggestion
1. Open Visual Studio.
```

01. Select the **Help** menu > **About Microsoft Visual Studio**.
01. Read the version number from the **About** dialog.

> ### Download

Reviewer: This section is more than just downloading VS

> Visual Studio installs the latest .NET Core SDK and runtime, so you can target your app to run on the .NET Core runtime version of your choice.

Reviewer: This sentence is not true.

- [Download Visual Studio](https://www.visualstudio.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=button+cta&utm_content=download+vs2019).

Selecting any of the managed workloads for **Web**, **Desktop**, or **Azure development**, will install the .NET Core SDK development tools and runtime. Specifically choosing the **.NET Core** workload also installs .NET Core.

[![Windows Visual Studio 2019 with .NET Core workload](media/install-sdk/windows-install-visual-studio-2019.png)](media/install-sdk/windows-install-visual-studio-2019.png#lightbox)

> After you've installed Visual Studio, create your first app by following the [C# Hello World tutorial](../tutorials/with-visual-studio.md) or the [Visual Basic Hello World tutorial](../tutorials/vb-with-visual-studio.md).

Reviewer: Shouldn't this be in the next steps instead?

::: zone-end

::: zone pivot="os-macos"

## Install with Visual Studio for Mac

> Visual Studio for Mac installs the .NET Core SDK when the **.NET Core workload** is selected. To get started with .NET Core development on macOS, see [Install Visual Studio 2019 for Mac](https://docs.microsoft.com/visualstudio/mac/installation?view=vsmac-2019).

Reviewer: You need to make the link relative to make it work with offline books.

```suggestion
Visual Studio for Mac installs the .NET Core SDK when the **.NET Core workload** is selected. To get started with .NET Core development on macOS, see [Install Visual Studio 2019 for Mac](/visualstudio/mac/installation).
```

After you've installed .NET Core support for VS Code, create your first app by following the [Using .NET Core in Visual Studio Code](https://code.visualstudio.com/docs/languages/dotnet) tutorial.

::: zone pivot="os-windows"

## Install with PowerShell automation

The [dotnet-install scripts](../tools/dotnet-install-script.md) are used for a non-admin install of the CLI toolchain and the shared runtime. You can download the script from [dotnet-install script reference page](../tools/dotnet-install-script.md).

The script defaults to installing the latest "LTS" version, which is currently .NET Core 2.1. To install the current release of .NET Core (3.0), run the script with the following switch:

```powershell
dotnet-install.ps1 -Channel 3.0
```

The installer PowerShell script is used in automation scenarios and non-admin installations.

::: zone-end

::: zone pivot="os-linux,os-macos"

## Install with bash automation

The [dotnet-install scripts](../tools/dotnet-install-script.md) are used for a non-admin install of the CLI toolchain and the shared runtime. You can download the script from [dotnet-install script reference page](../tools/dotnet-install-script.md).

> The script defaults to installing the latest "LTS" version, which is currently .NET Core 2.1. To install the current release of .NET Core, which is 3.0, run the script with the following switch:

Reviewer: you need to define what LTS is
